### ISSUE TITLE
Attention in float16 produces NaNs when using mask

### ISSUE THREAD

**Issue by myagues** [google/flax#982]
### Problem you have encountered:
Using attention + mask with `jnp.float16` produces NaNs.

### What you expected to happen:
Works correctly for `jnp.float32` and `jnp.bfloat16`. The problem might be when setting attention bias:
https://github.com/google/flax/blob/278df51cb97a4dcec05ab1b3b047e21bda72aa0b/flax/linen/attention.py#L233-L236

### Steps to reproduce:
[Colab link](https://colab.research.google.com/drive/17QAAJFoDUZtAcOtfqXrkPmTSVERFMC1i?usp=sharing)

**Comment by levskaya** [comment]
This would have been fixed by #1527, but was fixed by the more comprehensive gpu-training changes in #1414 .  Closing now.
